agent/engine/docker_task_engine_test.go

```diff
@@ -2694,3 +2695,83 @@ func TestTaskSecretsEnvironmentVariables(t *testing.T) {
 		})
 	}
 }
+
+// TestCreateContainerAddLogDriverConfig tests that in createContainer, when the
+// container's logConfig is set, the driver values are as expected
+func TestCreateContainerAddLogDriverConfig(t *testing.T) {
+	taskName := "logSenderTask"
+	taskARN := "logSenderTaskARN"
+	taskFamily := "logSenderTaskFamily"
+	taskVersion := "1"
+	logRouterType := "awslogrouter"
+	dataLogDriverPath := "/data/logrouter/"
+	dataLogDriverSocketPath := "/socket/fluent.sock"
+	rawHostConfigInput := dockercontainer.HostConfig{
+		LogConfig: dockercontainer.LogConfig{
+			Type: logRouterType,
+			Config: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+		},
+	}
+	rawHostConfig, err := json.Marshal(&rawHostConfigInput)
+	require.NoError(t, err)
+	getTask := func() *apitask.Task {
+		return &apitask.Task{
+			Arn:     taskARN,
+			Version: taskVersion,
+			Family:  taskFamily,
+			Containers: []*apicontainer.Container{
+				{
+					Name: taskName,
+					DockerConfig: apicontainer.DockerConfig{
+						HostConfig: func() *string {
+							s := string(rawHostConfig)
+							return &s
+						}(),
+					},
+				},
+			},
+		}
+	}
+
+	testCases := []struct {
+		name                           string
+		task                           *apitask.Task
+		expectedLogConfigType          string
+		expectedLogConfigTag           string
+		expectedLogConfigFluentAddress string
+	}{
+		{
+			name:                           "test create fluentd log router container",
+			task:                           getTask(),
+			expectedLogConfigType:          "fluentd",
+			expectedLogConfigTag:           taskName + "-" + taskARN,
+			expectedLogConfigFluentAddress: filepath.Join(defaultConfig.DataDirOnHost, dataLogDriverPath, taskARN, dataLogDriverSocketPath),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
since there's only one test case, it seems weird to have a for loop going through a list of test cases. Can you just write the test directly (without a loop and testCase struct)?

Reply:
Added a new one to test it doesn't populate logdriver in case the logconfig is not whats expected